component/src/main/java/org/wso2/extension/siddhi/io/cdc/source/CDCSource.java

```diff
@@ -0,0 +1,342 @@
+                                "\ndefine stream inputStream (id string, name string);",
+                        description = "In this example, the cdc source starts listening to the row insertions " +
+                                " on students table with columns name and id which is under MySQL" +
+                                " SimpleDB database that" +
+                                " can be accessed with the given url"
+                ),
+                @Example(
+                        syntax = "@source(type = 'cdc' , url = 'jdbc:mysql://localhost:3306/SimpleDB', " +
+                                "\nusername = 'cdcuser', password = 'pswd4cdc', " +
+                                "\ntable.name = 'students', operation = 'update', " +
+                                "\n@map(type='keyvalue', @attributes(id = 'id', name = 'name', " +
+                                "\nbefore_id = 'before_id', before_name = 'before_name')))" +
+                                "\ndefine stream inputStream (before_id string, id string, " +
+                                "\nbefore_name string, ," +
+                                " name string);",
+                        description = "In this example, the cdc source starts listening to the row updates" +
+                                " on students table which is under MySQL SimpleDB database that" +
+                                " can be accessed with the given url"
+                ),
+                @Example(
+                        syntax = "@source(type = 'cdc' , url = 'jdbc:mysql://localhost:3306/SimpleDB', " +
+                                "\nusername = 'cdcuser', password = 'pswd4cdc', " +
+                                "\ntable.name = 'students', operation = 'delete', " +
+                                "\n@map(type='keyvalue', @attributes(before_id = 'before_id'," +
+                                " before_name = 'before_name')))" +
+                                "\ndefine stream inputStream (before_id string, before_name string);",
+                        description = "In this example, the cdc source starts listening to the row deletions" +
+                                " on students table which is under MySQL SimpleDB database that" +
+                                " can be accessed with the given url"
+                )
+        }
+)
+
+public class CDCSource extends Source {
+    private static final Logger log = Logger.getLogger(CDCSource.class);
+    private final ExecutorService executorService = Executors.newSingleThreadExecutor();
+    private Map<byte[], byte[]> offsetData = new HashMap<>();
+    private String operation;
+    private ChangeDataCapture changeDataCapture;
+    private String historyFileDirectory;
+    private CDCSourceObjectKeeper cdcSourceObjectKeeper = CDCSourceObjectKeeper.getCdcSourceObjectKeeper();
+    private String carbonHome;
+
+    /**
```

Comment:
Remove inherited comments

Reply:
Fixed with commit: `Remove inherited comments in CDCSource.` d16fce1